torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -661,30 +696,63 @@ def _insert_or_replace_phs(new_args, name_suffix):
             false_graph, false_lifted_freevars, false_shared, unique_true, unique_false
         )
 
+        true_graph_module = torch.fx.GraphModule(true_nn_modules, true_graph)
+        false_graph_module = torch.fx.GraphModule(false_nn_modules, false_graph)
         true_name = add_subgraph(
             tx,
             self.source,
             "cond_true",
-            torch.fx.GraphModule(true_nn_modules, true_graph),
+            true_graph_module,
         )
         false_name = add_subgraph(
             tx,
             self.source,
             "cond_false",
-            torch.fx.GraphModule(false_nn_modules, false_graph),
+            false_graph_module,
         )
 
         true_node = make_attr(tx, true_name)
         false_node = make_attr(tx, false_name)
 
+        combined_args = true_shared + unique_true + unique_false
         p_args = (
             args[0].as_proxy(),
             true_node,
             false_node,
             # We pick true_shared but it shouldn't matter
-            true_shared + unique_true + unique_false,
+            combined_args,
         )
 
+        for branch_name, module in (
+            (True, true_graph_module),
+            (False, false_graph_module),
+        ):
+            try:
+                mutated_inputs, aliased_inputs = _detect_input_mutations_and_aliasing(
+                    tx, module, combined_args
+                )
+            except Exception as e:
+                unimplemented(
```

Comment:
try-except with bare Exception is dangerous, do we need this?

Reply:
Yeah, we can be more specific about the exception.